### PR TITLE
perf(ssz): remove bytes buffer from hasher

### PR DESCRIPTION
**Pull request by calbera**
## Summary by CodeRabbit

- **Refactor**
  - Unified and optimized cryptographic functions by migrating from the `sha256` package to the more generic `crypto` package.
  - Enhanced Merkle tree operations with efficient methods for root construction and parallelized hashing.

- **Tests**
  - Updated tests to align with the new `crypto` package for hashing operations.
  - Introduced new mock implementations for `HashFn` and `Hasher` types.

- **Chores**
  - Updated dependencies to include `github.com/klauspost/cpuid/v2` and `github.com/minio/sha256-simd`.

### CODE REVIEW

**Comment by itsdevbear** [comment]
despite this making sense, it feels weird how the function needs to take two "hasher things"

**Comment by calbera** [comment]
@itsdevbear i can put the functions on the Hasher but it feels like coupling two things that are not related

File: mod/primitives/pkg/crypto/sha256.go [hunk 1]
```diff
@@ -34,12 +34,12 @@ var sha256Pool = sync.Pool{New: func() interface{} {
...
 }}
 
-// Sum256 defines a function that returns the sha256 checksum of the data passed
+// Sha256 defines a function that returns the sha256 checksum of the data passed
 // in.
 // https://github.com/ethereum/consensus-specs/blob/v0.9.3/specs/core/0_beacon-chain.md#hash
 //
 //nolint:lll // url.
-func Sum256(data []byte) [32]byte {
+func Sha256(data []byte) [32]byte { // TODO: rename to Sha256
```
Comment (itsdevbear) [inline mod/primitives/pkg/crypto/sha256.go:42]: if anything call it Hash() cause rn it'll be sha256.Sha256() will be the usage.

Comment (calbera) [reply mod/primitives/pkg/crypto/sha256.go:42]: its `crypto.Sha256` now

Comment (itsdevbear) [inline mod/primitives/pkg/crypto/sha256.go:42]: tbh move it back to the sha256 package and name it Hash().

**Review by itsdevbear — COMMENTED** [review]
Nit, then LGTM

**Comment by itsdevbear** [comment]
did you sync against Bartio? all gucci?

**Review by itsdevbear — APPROVED** [review]

**Pull request merged**